src/gpu/vulkan/SDL_gpu_vulkan.c

```diff
@@ -703,7 +703,7 @@ typedef struct WindowData
 
     // Synchronization primitives
     VkSemaphore imageAvailableSemaphore[MAX_FRAMES_IN_FLIGHT];
```

Comment:
This should probably also use the image count.

Reply:
That's what I kind of expected at first as well, but the example in the Vulkan docs explicitly says only the submit semaphore needs this.
```
    // Index submit semaphore with the acquired swapchain image index.
    // It's the only resource in this example indexed by image_index.
    // All other resources, including acquire_semaphore, are indexed with current in-flight frame index.
```
It makes sense though since you could be waiting to acquire up to frames in flight count, but that is independent of what has been submitted but may not have yet been presented.  Also vkAcquireNextImageKHR is what gives you the image index but you need to pass a semaphore to it.